src/model/chunk.py

```diff
@@ -0,0 +1,36 @@
+## This file contains the function to chunk the input text into smaller pieces for the model to process
+## The input text is tokenized and then split into chunks of size chunk_size
+
+from transformers import AutoTokenizer
+import torch
+
+def chunk(inp: str, 
+          tokenizer: AutoTokenizer, 
+          chunk_size: int = 256, 
+          overlapping_len: int = 3,
+          num_chunks: int = 128) -> list:
+    tokenized_txt = tokenizer(inp, return_tensors="pt")["input_ids"]
+    token_len = tokenized_txt.size()[1]
+    
+    if token_len <= chunk_size:
+        return [tokenized_txt]
```

Comment:
wouldn't we want to pad in the `token_len < chunk_size` case?

Reply:
Resolved by using built-in padding of ```AutoTokenizer``` in https://github.com/MichiganDataScienceTeam/F24-mini-copilot/pull/24/commits/926beedac4e3a431f36582d81dae250ddf5e172f